perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -803,6 +826,81 @@ public void moveGroup(PerunSession sess, Group destinationGroup, Group movingGro
 		}
 
 		getPerunBl().getAuditer().log(sess, new GroupMoved(movingGroup));
+
+		// Check if moved group should be autoassigned to destination's group resource
+		if (destinationGroup != null) {
+			autoassignMovedTree(sess, destinationGroup, movingGroup);
+		}
+	}
+
+	/**
+	 * Checks, if moved group and subgroups should be automatically assigned to any group
+	 * @param sess
+	 * @param destinationGroup
+	 * @param movingGroup
+	 * @throws WrongReferenceAttributeValueException
+	 * @throws WrongAttributeValueException
+	 */
+	private void autoassignMovedTree(PerunSession sess, Group destinationGroup, Group movingGroup) throws WrongReferenceAttributeValueException, WrongAttributeValueException {
+		List<AssignedResource> resourcesToAutoassign = perunBl.getResourcesManagerBl().getResourceAssignments(sess, destinationGroup, List.of()).stream()
+			.filter(AssignedResource::isAutoAssignSubgroups)
+			.collect(toList());
+
+		for (AssignedResource resourceToAutoassign : resourcesToAutoassign) {
+			Group sourceGroup;
+			try {
+				sourceGroup = resourceToAutoassign.getSourceGroupId() == null ? destinationGroup : this.getGroupById(sess, resourceToAutoassign.getSourceGroupId());
+			} catch (GroupNotExistsException e) {
+				throw new ConsistencyErrorException(e);
+			}
+
+			List<Group> groupsToAutoAssign = perunBl.getGroupsManagerBl().getAllSubGroups(sess, movingGroup);
+			groupsToAutoAssign.add(movingGroup);
+
+			for (Group groupToAutoassign : groupsToAutoAssign) {
+				try {
+					perunBl.getResourcesManagerBl().assignAutomaticGroupToResource(sess, sourceGroup, groupToAutoassign, resourceToAutoassign.getEnrichedResource().getResource(),
+						resourceToAutoassign.getStatus().equals(GroupResourceStatus.INACTIVE));
+				} catch (GroupAlreadyAssignedException e) {
+					// skip
+				} catch (GroupResourceMismatchException e) {
+					log.error("Could not autoassign group " + groupToAutoassign + " to resource " + resourceToAutoassign, e);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Checks, if moving group would still belong under source group tree of automatic assignments on all assigned resources
+	 * and removes together with subgroups from source group's autoassignments if not
+	 * @param sess
+	 * @param destinationGroup
+	 * @param movingGroup
+	 */
+	private void fixMovedTreeAutoassignments(PerunSession sess, Group destinationGroup, Group movingGroup) {
+		List<AssignedResource> autoAssignedResources = perunBl.getResourcesManagerBl().getResourceAssignments(sess, movingGroup, List.of()).stream()
+			.filter(g -> g.getSourceGroupId() != null)
+			.collect(toList());
+
+		for (AssignedResource autoAssignedResource : autoAssignedResources) {
+			int sourceGroupId = autoAssignedResource.getSourceGroupId();
+			try {
+				Group sourceGroup = this.getGroupById(sess, sourceGroupId);
+				List<Group> groupsToRemove = this.getAllSubGroups(sess, sourceGroup);
```

Comment:
```suggestion
				List<Group> groupsToRemove = this.getAllSubGroups(sess, movingGroup);
```
I think we should remove autoassignments of moving group's subgroups. The following if condition should be changed as well.

Reply:
Right, it was wrong tree to remove. Now the check is the same - if destination group is subgroup of source group, then it creates new tree starting from the moving group.